wcfsetup/install/files/lib/acp/form/CaptchaQuestionAddForm.class.php

```diff
@@ -46,155 +37,76 @@ class CaptchaQuestionAddForm extends AbstractForm
-
-        I18nHandler::getInstance()->register('question');
-        I18nHandler::getInstance()->register('answers');
-    }
-
-    /**
-     * @inheritDoc
-     */
-    public function save()
-    {
-        parent::save();
-
-        $this->objectAction = new CaptchaQuestionAction([], 'create', [
-            'data' => \array_merge($this->additionalFields, [
-                'answers' => I18nHandler::getInstance()->isPlainValue('answers') ? I18nHandler::getInstance()->getValue('answers') : '',
-                'isDisabled' => $this->isDisabled,
-                'question' => I18nHandler::getInstance()->isPlainValue('question') ? I18nHandler::getInstance()->getValue('question') : '',
-            ]),
+        parent::createForm();
+
+        $this->form->appendChildren([
+            FormContainer::create('data')
+                ->appendChildren([
+                    TextFormField::create('question')
+                        ->label('wcf.acp.captcha.question.question')
+                        ->maximumLength(255)
+                        ->required()
+                        ->i18n()
+                        ->languageItemPattern('wcf.captcha.question.question.question\d+'),
+                    MultilineTextFormField::create('answers')
+                        ->label('wcf.acp.captcha.question.answers')
+                        ->required()
+                        ->i18n()
+                        ->languageItemPattern('wcf.captcha.question.answers.question\d+')
+                        ->addValidator(new FormFieldValidator('regex', function (MultilineTextFormField $formField) {
+                            if ($formField->getValidationErrors() !== []) {
+                                return;
+                            }
+
+                            // returns `true` for a valid answer
+                            $answerValidator = static function (string $text): bool {
+                                if (!\mb_substr($text, 0, 1) == '~' || !\mb_substr($text, -1, 1) == '~') {
+                                    return true;
+                                }
+
+                                $regexLength = \mb_strlen($text) - 2;
+
+                                return $regexLength && Regex::compile(\mb_substr($text, 1, $regexLength))->isValid();
+                            };
+
+                            if ($formField->hasI18nValues()) {
+                                $answers = [];
+                                foreach ($formField->getValue() as $value) {
+                                    $answers = \array_merge($answers, \explode("\n", StringUtil::unifyNewlines($value)));
+                                }
+                            } else {
+                                $answers = \explode("\n", StringUtil::unifyNewlines($formField->getValue()));
+                            }
+
+                            foreach ($answers as $answer) {
+                                if (!$answerValidator($answer)) {
```

Comment:
```suggestion
                                if (!$this->validateAnswer($answer)) {
```
see previous comment.